### PR TITLE
76685 represented organisation details tab

### PR DESCRIPTION
**Pull request by stevelee-civica**
76685 represented organisation details tab

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Producer/Controllers/ProducerController.cs [hunk 1]
```diff
@@ -189,10 +190,17 @@ public async Task<ActionResult> RepresentedOrganisationDetails()
...
+            {
+                SmallProducerSubmissionData = SmallProducerSubmissionData
+            };
+
             var vm = new OrganisationDetailsTabsViewModel
             {
                 ActiveOption = OrganisationDetailsDisplayOption.RepresentedOrganisationDetails,
-                SmallProducerSubmissionData = this.SmallProducerSubmissionData
+                SmallProducerSubmissionData = this.SmallProducerSubmissionData,
+                RepresentingCompanyDetailsViewModel = SmallProducerSubmissionData.HasAuthorisedRepresentitive ?
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Controllers/ProducerController.cs:202]: @stevelee-civica  this wont work as the mapper is currently mapping the current submission values, we need to map what has been saved to the high level organisation data.
We need to do what we discussed this morning
image

Comment (stevelee-civica) [reply src/EA.Weee.Web/Areas/Producer/Controllers/ProducerController.cs:202]: updated

Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Producer/Controllers/ProducerController.cs:202]: @stevelee-civica  I can't see any change to the GetSmallProducerSubmissionHandler

Comment (stevelee-civica) [reply src/EA.Weee.Web/Areas/Producer/Controllers/ProducerController.cs]: pushed to please try again

File: src/EA.Weee.Web/Areas/Producer/Controllers/ProducerController.cs [hunk 2]
```diff
@@ -195,12 +195,13 @@ public async Task<ActionResult> RepresentedOrganisationDetails()
...
 
+            var representingCompanyDetailsViewModel = mapper.Map<SmallProducerSubmissionMapperData, RepresentingCompanyDetailsViewModel>(source);
+
             var vm = new OrganisationDetailsTabsViewModel
             {
                 ActiveOption = OrganisationDetailsDisplayOption.RepresentedOrganisationDetails,
                 SmallProducerSubmissionData = this.SmallProducerSubmissionData,
-                RepresentingCompanyDetailsViewModel = SmallProducerSubmissionData.HasAuthorisedRepresentitive ? 
-                    mapper.Map<SmallProducerSubmissionMapperData, RepresentingCompanyDetailsViewModel>(source) : null
+                RepresentingCompanyDetailsViewModel = representingCompanyDetailsViewModel
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Controllers/ProducerController.cs:204]: @stevelee-civica  looking at the changes now. Do we actually need this RepresentingCompanyDetailsViewModel on the viewmodel. We can access the rep company data through SmallProducerSubmissionData.AuthorisedRepresentitiveData etc...

File: src/EA.Weee.Web/Areas/Producer/Views/Producer/ViewOrganisation/RepresentedOrganisationDetails.cshtml [hunk 3]
```diff
@@ -14,12 +14,60 @@
         @Html.Partial("ViewOrganisation/_OrganisationDetailsTabs", Model)
 
         <div class="govuk-tabs__panel" id="organisationDetails">
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Views/Producer/ViewOrganisation/RepresentedOrganisationDetails.cshtml:16]: @stevelee-civica  do we need to conditionally show this when its an authed rep or is it handled elsewhere?

Comment (stevelee-civica) [reply src/EA.Weee.Web/Areas/Producer/Views/Producer/ViewOrganisation/RepresentedOrganisationDetails.cshtml:16]: inside _OrganisationDetailsTabs.cshtml

File: src/EA.Weee.Web.Tests.Unit/Areas/Producer/Mapping/ToViewModel/ServiceOfNoticeMapTests.cs [hunk 4]
```diff
@@ -65,6 +67,7 @@ public void Map_ShouldMapNullServiceOfNoticeAddress()
         {
             // Arrange
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Areas/Producer/Mapping/ToViewModel/ServiceOfNoticeMapTests.cs:68]: @stevelee-civica  might need to test the true case here :)

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/ServiceOfNoticeMap.cs [hunk 5]
```diff
@@ -23,7 +23,7 @@ public ServiceOfNoticeViewModel Map(SmallProducerSubmissionMapperData source)
             {
                 DirectRegistrantId = submissionData.DirectRegistrantId,
                 OrganisationId = submissionData.OrganisationData.Id,
-                Address = source.UseMasterVersion ? MapAddress(submissionData.ServiceOfNoticeData) : MapAddress(submissionData.CurrentSubmission.ServiceOfNoticeData),
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/ServiceOfNoticeMap.cs]: @stevelee-civica this line Address = source.UseMasterVersion ? MapAddress(submissionData.ServiceOfNoticeData) : MapAddress(submissionData.CurrentSubmission.ServiceOfNoticeData), looks correct to me

**Pull request closed without merging**